### ISSUE TITLE
False positive for `no-array-callback-reference` when passing an object into the MongoDB's `.find()` function

### ISSUE THREAD

**Issue by Miodec** [sindresorhus/eslint-plugin-unicorn#1940]
`no-array-callback-reference`

My code builds an object which is then used as a parameter to the MongoDB's `.find()` function. This object is then mislabeled as a "function" and the MongoDB's `.find()` is misinterpreted as `Array.find()`

```ts
export async function get(language: string): Promise<MonkeyTypes.NewQuote[]> {
  if (!git) throw new MonkeyError(500, "Git not available.");
  const where: {
    approved: boolean;
    language?: string;
  } = {
    approved: false,
  };
  if (language !== "all") {
    where.language = language;
  }
  return await db
    .collection<MonkeyTypes.NewQuote>("new-quotes")
    .find(where)
    .sort({ timestamp: 1 })
    .limit(10)
    .toArray();
}
```

<img width="817" alt="image" src="https://user-images.githubusercontent.com/13181393/198149992-c405d7c3-bd1b-4638-b7e2-db2be14d8dec.png">

**Comment by fisker** [comment]
There's no way for the rule to know it's not an array from just static checking. You'll have to disable the rule.
